### PR TITLE
Support restore in restoring mode and merging mode

### PR DESCRIPTION
**Pull request by zhoney**
Change-Id: Ie8dcf37cbef696d4acd5b9faf32d703fb8a3f5c4

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/cmd/SubCommands.java [hunk 1]
```diff
@@ -636,6 +635,22 @@ public void validate(String name, String value) {
         }
     }
 
+    public static class RestoreModeValidator implements IParameterValidator {
+
+        private static final Set<String> RESTORE_MODES = ImmutableSet.of(
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/cmd/SubCommands.java]: defined in client

File: src/main/java/com/baidu/hugegraph/manager/RestoreManager.java [hunk 2]
```diff
@@ -49,10 +50,19 @@
...
 
     private static final int BATCH_SIZE = 500;
+    private String mode = null;
+    private static final String MERGING = "MERGING";
+    private static final String RESTORING = "RESTORING";
+    public static final Set<String> MODES = ImmutableSet.of(
+            RESTORING, MERGING
+    );
+
+    private static Map<String, Long> primaryKeyVLs = null;
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/RestoreManager.java]: don't set static due to not shared by multi graphs

**Review by javeme — CHANGES_REQUESTED** [review]

File: src/main/java/com/baidu/hugegraph/cmd/HugeGraphCommand.java [hunk 3]
```diff
@@ -114,10 +115,19 @@ private void execute(String subCmd, JCommander jCommander) {
                 backupManager.backup(backup.types(), backup.directory());
                 break;
             case "restore":
-                Printer.print("Graph '%s' start restore!", this.graph());
+                SubCommands.GraphModeGet graphModeGet =
+                            this.subCommand("graph-mode-get");
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/cmd/HugeGraphCommand.java]: align with graphModeGet

File: src/main/java/com/baidu/hugegraph/cmd/SubCommands.java [hunk 4]
```diff
@@ -455,12 +456,13 @@ public int retry() {
...
     }
 
-    public class RestoreFlag {
+    public class Mode {
 
-        @Parameter(names = {"--restore", "-r"}, arity = 1,
-                   description = "Restore flag",
+        @Parameter(names = {"--graph-mode", "-m"}, arity = 1,
+                   converter = GraphModeConverter.class,
+                   description = "Graph mode, [NONE, RESTORING, MERGING]",
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/cmd/SubCommands.java]: "Graph mode, include: [NONE, RESTORING, MERGING]

File: src/main/java/com/baidu/hugegraph/manager/RestoreManager.java [hunk 5]
```diff
@@ -250,4 +271,33 @@ private void restore(HugeType type, File file,
...
+        this.primaryKeyVLs = new HashMap<>();
+        List<VertexLabel> vertexLabels = this.client.schema().getVertexLabels();
+        for (VertexLabel vl : vertexLabels) {
+            if (vl.idStrategy() == IdStrategy.PRIMARY_KEY) {
+                this.primaryKeyVLs.put(vl.name(), vl.id());
+            }
+        }
+    }
+
+    private void processVertexIdInEdge(List<Edge> edges) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/RestoreManager.java]: updateVertexIdInEdge

File: src/main/java/com/baidu/hugegraph/manager/RestoreManager.java [hunk 6]
```diff
@@ -137,6 +141,7 @@ private void restoreEdges(HugeType type, String dir) {
             for (int i = 0; i < size; i += BATCH_SIZE) {
                 int toIndex = Math.min(i + BATCH_SIZE, size);
                 List<Edge> subEdges = edges.subList(i, toIndex);
+                processVertexIdInEdge(subEdges);
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/RestoreManager.java]: rename to updateVertexIdInEdge
and comment why needed

**Review by javeme — CHANGES_REQUESTED** [review]

**Review by javeme — APPROVED** [review]

**Review by Linary — APPROVED** [review]

**Pull request merged**